File: compiler/testData/loadJava/kotlinSignature/propagation/return/TwoSuperclassesReturnSameJavaType.java
package test;

import org.jetbrains.annotations.NotNull;
import java.lang.CharSequence;

import jet.runtime.typeinfo.KotlinSignature;

public interface TwoSuperclassesReturnSameJavaType {
    public CharSequence foo();

    public interface Other {
        @NotNull
        public CharSequence foo();
    }

    public interface Sub extends TwoSuperclassesReturnSameJavaType, Other {
        public CharSequence foo();
    }
}

File: compiler/testData/loadJava/kotlinSignature/propagation/return/TwoSuperclassesReturnSameJavaType.kt
package test

public trait TwoSuperclassesReturnSameJavaType: Object {

    public trait Super1: Object {
        public fun foo(): CharSequence?
    }

    public trait Super2: Object {
        public fun foo(): CharSequence
    }

    public trait Sub: Super1, Super2 {
        override fun foo(): CharSequence
    }
}